translations/test/ElementMergeServer.js

```diff
@@ -81,13 +81,13 @@ var areasInput =
 var buildingsInput =
     "<?xml version='1.0' encoding='UTF-8'?>\
      <osm version='0.6' generator='JOSM'>\
-       <way id='-222330' action='modify' visible='true'>\
-         <nd ref='-222258' />\
-         <nd ref='-222256' />\
-         <nd ref='-222254' />\
-         <nd ref='-222252' />\
-         <nd ref='-222264' />\
-         <nd ref='-222258' />\
+       <way id='222330' action='modify' visible='true'>\
+         <nd ref='222258' />\
+         <nd ref='222256' />\
+         <nd ref='222254' />\
+         <nd ref='222252' />\
+         <nd ref='222264' />\
+         <nd ref='222258' />\
```

Comment:
I changed the signs for the element with the `hoot:merge:target` tag to be positive.  The way that the mergers work is that when the signs of the IDs match (negative and negative or positive and positive) the merger follows the direction of the merge target tag and merges into that element.  When they differ, the mergers keep the element with the positive ID is the element ID that is retained.  This is the product of non-destructive conflation where positive IDs (or existing IDs) are favored over negative (or new) IDs.

So the fix in the CPP code will ensure that there won't be a crash anymore if the element with the negative ID is the target and the positive ID will be preserved in the merge.  The tests should run clean now.

Reply:
I think we need a test that exercises the positive id merged into negative and checks that the positive id is preserved.  I'm thinking this would also have to update the negative id way nodes to re-use the node ids of the positive id way that's being removed after the merge.

Reply:
After a brief inspection, the ui code just copies the tags from the merged feature.  So the id swapping to reuse the positive id is just ignored.

I will still get test added.